File: VB/WinSolution.Module/DemoObject.vb
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp

Namespace WinSolution.Module
	<DefaultClassOptions, DefaultListViewOptions(True, NewItemRowPosition.Top)> _
	Public Class DemoObject
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _Name As String
		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _Name, value)
			End Set
		End Property
		Private _LookupProperty As DemoLookupObject
		<ImmediatePostData, VisibleInListView(True)> _
		Public Property LookupProperty() As DemoLookupObject
			Get
				Return _LookupProperty
			End Get
			Set(ByVal value As DemoLookupObject)
				SetPropertyValue("LookupProperty", _LookupProperty, value)
			End Set
		End Property
	End Class
End Namespace

File: CS/XPO/WinSolution.Module/DemoObject.cs
using System;
using DevExpress.Xpo;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp;

namespace WinSolution.Module {
    [DefaultClassOptions]
    [DefaultListViewOptions(true, NewItemRowPosition.Top)]
    public class DemoObject : BaseObject {
        public DemoObject(Session session) : base(session) { }
        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }
        private DemoLookupObject _LookupProperty;
        [ImmediatePostData]
        [VisibleInListView(true)]
        public DemoLookupObject LookupProperty {
            get { return _LookupProperty; }
            set { SetPropertyValue("LookupProperty", ref _LookupProperty, value); }
        }
    }
}
